### ISSUE TITLE
Integration with livestreamer

### ISSUE THREAD

**Issue by john-banks** [metal3d/idok#4]
Livestreamer has a whole host of media it can stream but can't go to xbmc/kodi. I was able to take the json livestreamer produces and feed it into idok. Was thinking it'd be useful to have idok make use of livestreamers json api to give it access to all its available streams.

**Comment by metal3d** [comment]
Hi, excuse my english but I'm not sure to understand what you mean. 
What you mean is that idok should reconnize livestream urls, get json api to get the stream url then send it to Kodi ? Right ?
Actually, that's not a bad idea (if I understand what you ask). I do it with Youtube streams (using youtube plugin url to kodi).
I'm planning to create a "plugin system" that will be able to reconnize some stream url and do tasks to send command to Idok. 
If I didn't understood your question, please correct me :)

**Comment by john-banks** [comment]
Yep that's exactly what I mean. Livestreamer already supports a whole load of different video sites so it would simplfy sending things to kodi.

**Comment by metal3d** [comment]
I'm sorry but... are you speaking about livestreamer python tool ? or the website ? Because with the python tool, you (at this time) send it to stdin, right ?
Anyway, I will create a plugin system :)

**Comment by curunoir** [comment]
Hi, I'm also interested in this request. I used this little project https://github.com/athoik/livestreamersrv/tree/master
to integrate livestreamer in a Serviio plugin, a dlna server, and then I played movies from Xbmc using the upnp browser. But a more direct approach as talked here would fit better for me.

**Comment by curunoir** [comment]
Well I've got it working ! Thanks it's a great project. Now I'm thinking at how we can invoke idok directly from Kodi without using the CLI, maybe with this project https://github.com/pdf/kodi-callback-daemon
What do you think ?

**Comment by metal3d** [comment]
Thanks for you thanks ;)

At this time, I spend a lot of time on another project so I don't finalized idok yet.

To clarify your demand, you want to invoke idok from kodi, but how ? from the kodi menu ?

That's not, IMHO, the goal of idok. Idok is made to stream media from computer to Kodi via your computer. If you want to stream from computer to kodi using kodi interface, you can use DLNA server. 

I'm using rygel on my Linux boxes (I should open ports and all...) that is very well integrated in system. But that's not exactly the same approach of idok.

idok allows me to stream other things from my computer than movies, photos or music. It's a basic way to make a "one shot streaming" process.

But ! let me think about what you're saying: call idok from kodi. If you're thinking about installing idok on the kodi system, it's possible. Because I develop idok with Go, I can compile for several CPU architectures. And... I tried ! It works like a charm. The idea is to send command like that:

    idok /path/to/file -target=localhost 

As "localhost" means "myself", kodi connect itself to itself and it's working !

So, what I can try is to develop an addon that installs idok, set some connect point (localfile, a configured local computer, or livestreamer for example) then make the addon launching idok commands.

That's not impossible, just need some time :)

At this time, I've several things to do:

- discover kodi with mdns requests (avahi, bonjour, and so on)
- fix some issues
- finalize compilation process to serve a good new version
- set up a "check update" functionnality

Afterward, I will be able to continue on plugins, maybe addons...

**Comment by curunoir** [comment]
Hi, thanks for your feedback. (Vu qu'on a l'air tout les deux français et que mon anglais est franchement bof je me demande si on ne peut pas continuer ici en fr ou par mail)

**Comment by curunoir** [comment]
To explain my needs, I own a TV Box running Kodi on Android. And I wanted to see Dailymotion live streams from it, since the Dailymotion didn't work on my box. But I discovered livestreamer and used it on a remote machine to serve these streams through DLNA with the livestreamer server daemon and a Serviio plugin I wrote. That worked, sometimes.

Since the device running Kodi is on Android I can't easily install livestreamer nor idok on this box. So that's why I run them on a remote machine with "standard" Linux. It is working and I can now play these streams on my Kodi box by invoking idok and livestreamer on this remote machine. But in the ideal world I would like to start the playback directly from Kodi, typically within a video Kodi addon or maybe from Kodi favourites, I hope this explains better my demand.

Also i'm more of a Php/Java developper but Kodi addon development is interesting me so I would be glad to give some help.

**Comment by metal3d** [comment]
(héhé oui je suis français, on peut en parler par mail si tu le souhaite, mais comme je commence à avoire des retours sur l'appli, le fait d'en parler dans une issue permet aux autres d'avoir l'info)

Ok, back to english :)

What you need, in my point of view, is clearly to be able to install idok on you media box (raspberry ?) and to be able to control it from kodi menu. I will take a look on what I can do for this. 

Addons are made in Python - and that's one of my 2 favorite laguage (with Go). I'll see what I can do on Saturday.

(metal3d@gmail.com)

**Comment by curunoir** [comment]
My media is stored on a NAS and my media player box is Android, I don't know well if it easy to get it running idok or even livestreamer and its dependencies : ffmpeg...
